src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -642,6 +642,19 @@ int xdg_autostart_service_generate_unit(
                         e_not_show_in);
         }
 
+        /* Generate an ExecCondition for KDE autostart condition */
+        if (!isempty(service->kde_autostart_condition)) {
+                _cleanup_free_ char *kde_autostart_condition_path = NULL;
+                if (find_binary("kde-systemd-start-condition", &kde_autostart_condition_path) < 0) {
+                        fprintf(f,
+                                "ExecCondition=%s \"%s\"",
+                                kde_autostart_condition_path,
+                                service->kde_autostart_condition);
+                } else {
+                        log_warning_errno(r, "kde-systemd-start-condition not found: %m");
```

Comment:
We should probably generate something that prevents execution. Maybe even just `ExecCondition=/bin/false` …

Reply:
In the non-systemd world if you have something with

OnlyShowIn=KDE,Gnome
X-KDE-autostart-condition=blah

and you log into gnome it'll be executed.

Which is the logic being somewhat-duplicated here. 

Maybe we should go all-out on that and check XDG_SESSION_DESKTOP=KDE  instead.

Reply:
What do you mean? We should try to only evaluate `X-KDE-autostart-condition` in KDE and `AutostartCondition` in GNOME?

Though, I guess that would mean knowing that we are actually managed by the corresponding desktop environment. I fear too many people abuse GNOME to actually be sure that if `XDG_CURRENT_DESKTOP` does not contain GNOME it actually isn't running gnome-session.

But, you could just duplicate the `xdg-autostart-condition` check that systemd already does. i.e. always return success if `XDG_CURRENT_DESKTOP` contains KDE.

Reply:
My concern was that the unit would fail to start if KDE isn't installed and the binary isn't present. The `X-KDE-autostart-condition` property could be set on a desktop file that isn't from KDE

Reply:
Yeah, we should probably just leave it as is. Then the KDE binary can decide on the policy if e.g. started under GNOME (if it can detect that …).